src/mjolnir/graphbuilder.cc

```diff
@@ -159,7 +160,7 @@ void ConstructEdges(const OSMData& osmdata,
     // Remember this edge starts here
     Edge prev_edge = Edge{0};
     Edge edge = Edge::make_edge(way_node.way_index, current_way_node_index, way);
-    edge.attributes.way_begin = true;
+    edge.attributes.way_begin = way_node.way_shape_node_index == 0;
```

Comment:
we could go either way on this. @gknisely what is preferable, i know this has implications for lane stuff

Reply:
without testing I am not sure.